command/debug_test.go

```diff
@@ -0,0 +1,104 @@
+package command
+
+import (
+	"os"
+	"path/filepath"
+	"testing"
+	"time"
+
+	"github.com/mitchellh/cli"
+	"github.com/stretchr/testify/require"
+)
+
+func TestDebugUtils(t *testing.T) {
+	xs := argNodes("foo, bar")
+	require.Equal(t, []string{"foo", "bar"}, xs)
+
+	xs = argNodes("")
+	require.Len(t, xs, 0)
+	require.Empty(t, xs)
+}
+
+func TestDebugFails(t *testing.T) {
+	t.Parallel()
+	srv, _, _ := testServer(t, false, nil)
+	defer srv.Shutdown()
+
+	ui := new(cli.MockUi)
+	cmd := &DebugCommand{Meta: Meta{Ui: ui}}
+
+	// Fails incorrect args
+	code := cmd.Run([]string{"some", "bad", "args"})
+	require.Equal(t, 1, code)
+
+	// Fails illegal node ids
+	code = cmd.Run([]string{"-node-id", "foo:bar"})
+	require.Equal(t, 1, code)
+
+	// Fails missing node ids
+	code = cmd.Run([]string{"-node-id", "abc,def"})
+	require.Equal(t, 1, code)
+
+	// Fails bad durations
+	code = cmd.Run([]string{"-duration", "foo"})
+	require.Equal(t, 1, code)
+
+	// Fails bad durations
+	code = cmd.Run([]string{"-interval", "bar"})
+	require.Equal(t, 1, code)
+
+	// Fails existing output
+	format := "2006-01-02-150405Z"
+	stamped := "nomad-debug-" + time.Now().UTC().Format(format)
+	path := filepath.Join(os.TempDir(), stamped)
+	os.MkdirAll(path, 0755)
+	defer os.Remove(path)
+	code = cmd.Run([]string{"-output", os.TempDir()})
+	require.Equal(t, 2, code)
+}
+
+func TestDebugCapturedFiles(t *testing.T) {
+	t.Parallel()
+	srv, _, _ := testServer(t, false, nil)
+	defer srv.Shutdown()
+
+	ui := new(cli.MockUi)
+	cmd := &DebugCommand{Meta: Meta{Ui: ui}}
+
+	// Calculate the output name
+	format := "2006-01-02-150405Z"
+	stamped := "nomad-debug-" + time.Now().UTC().Format(format)
+	path := filepath.Join(os.TempDir(), stamped)
+	defer os.Remove(path)
+
+	code := cmd.Run([]string{
+		"-output", os.TempDir(),
+		"-server-id", "leader",
+		"-duration", "1s",
+		"-interval", "500ms",
+	})
+	require.Equal(t, 0, code)
+
+	// Version is always captured
+	require.FileExists(t, filepath.Join(path, "version", "agent-self.json"))
+
+	// Consul and Vault are only captured if they exist
+	_, err := osStat(filepath.Join(path, "version", "consul-agent-self.json"))
```

Comment:
This is a testing gap, especially because there are couple of variations of getting tokens in and http vs. https. I think the most straightforward way to test this is in the e2e cluster where we have a running consul and vault.